packages/@uppy/dashboard/src/components/AddFiles.js

```diff
@@ -31,7 +19,7 @@ class AddFiles extends Component {
     event.target.value = null
   }
 
-  renderPoweredByUppy () {
+  renderPoweredByUppy = () => {
```

Comment:
Are all these changes necessary? :thinking:

Reply:
Yeah, but if we always use class properties, we never have to think about binds? Are there downsides, extra code in the transpiled output?

Reply:
I can think of one though it isn't super likely to affect us…class properties are properties of the instance, not of the prototype, and instance properties override the prototype.

```js
class A {
  getName = () => 'A'
}
class B extend A {
  getName () { return 'B' }
}

new B().getName()
// returns 'A'
```

First a new object with `B`'s prototype is created, then `A`'s implicit constructor runs, and that constructor assign `this.getName = () => 'A'`.

`.bind()` doesn't have this problem because it uses the function on the prototype: `this.getName.bind(this)` still refers to `B`'s implementation there.

I think the only places where this could affect us is inside the `Plugin` base class and in provider views (because google drive has a class that inherits from `ProviderViews`), so we can just decide that the risk is negligible :woman_shrugging: I normally use method syntax for things that are methods (like `render`), and class properties for things like event handlers.